code/modules/mob/living/simple_animal/hostile/mining/goliath.dm

```diff
@@ -195,3 +252,329 @@
+			if(food_wanted == ORGANS)
+				msgs += "<span class='notice'>It seems to want to eat something soft!</span>"
+			if(food_wanted == DIAMOND)
+				msgs += "<span class='notice'>It seems to want to eat something crunchy!</span>"
+		. += msgs.Join("<BR>")
+
+/mob/living/simple_animal/hostile/asteroid/goliathjuv/Life()
+	..()
+	if(stat != DEAD)
+		growth++
+		if(tame_progress >= 1 && tamed != 2) // Lose taming progress with time, unless you've been tamed
+			tame_progress--
+		if(feed_cooldown >= 1) // Lower the cooldown to be fed
+			feed_cooldown--
+		handle_preattack()
+		if(tame_progress <= 599 && tamed != 0) // Become feral if left alone for too long
+			tamed = 0
+			visible_message("<span class='warning'>[src] looks frenzied!</span>")
+			food_wanted = MEAT
+		if(tame_progress >= 600 && tamed == 0) // Become neutral
+			tamed = 1
+			visible_message("<span class='warning'>[src] looks pacified...</span>")
+			reroll_food()
+		if(tame_progress >= 2400 && !picking_candidates && !key) // Become player controlled
+			picking_candidates = TRUE
+			becomeaware()
+		if(tame_progress >= 2400 && key && tamed == 1)
+			to_chat(src, "<span class='biggerdanger'>You have been tamed!</span>")
+			to_chat(src, "<span class='danger'>You have been, by being fed and treated well, introduced to the non-aggressive way of living, but you have also been disowned by the creatures you previously coexisted with.</span>")
+			tamed = 2
+			creator.juvgoliathsspawned--
+			creator = null
+	if(tamed == 2)
+		faction = list("neutral")
+	if(tamed == 1)
+		faction |= "neutral"
+	if(tamed == 0)
+		faction = list("mining")
+	if(stat != DEAD && (getBruteLoss() || getFireLoss())) // Due to being young, their cells multiply at a quicker rate, allowing them to heal
+		adjustBruteLoss(-1)
+		adjustFireLoss(-1)
+		if(prob(70) && tamed != 2 && tame_progress >= 1) // Lose taming progress if you were hurt
+			tame_progress--
+	if(!target && creator && prob(70) && stat != DEAD && creator.stat != DEAD && tamed == 0) // Stick around your ancient goliath if you're not chasing after anything and aren't being tamed
+		var/turf/T = get_turf(creator)
+		var/list/surrounding_turfs = block(locate(T.x - 1, T.y - 1, T.z), locate(T.x + 1, T.y + 1, T.z))
+		if(!surrounding_turfs.len)
+			return
+		if(get_dist(src, T) <= 6)
+			return
+		if(isturf(src.loc) && get_dist(src, T) <= 50)
+			LoseTarget()
+			Goto(pick(surrounding_turfs), move_to_delay)
+			return
+	if(growth >= 600 && subadult == 0 && stat != DEAD) // Grow to subadult
+		name = "subadult goliath"
+		subadult = 1
+		maxHealth = maxHealth += 50
+		health = health += 50
+		speed = 2
+		move_to_delay = 4
+		desc = "A medium-sized red animal."
+		speak_emote = list("growls")
+		icon_state = "goliath_subadult"
+		icon_living = "goliath_subadult"
+		icon_dead = "goliath_subadult_dead"
+		ranged = TRUE
+		environment_smash = 2
+		add_draconian_effect()
+	if(growth >= 1200 && subadult == 1 && stat != DEAD) // Grow to adult
```

Comment:
Is there ever a situation where growth is this high but subadult isn't set? Also, do you really need to check if you're dead? Is Life() still called on dead mobs?

Reply:
Life is being called on dead mobs I believe after a quick check

Reply:
The first situation can happen with admin intervention, while the second case actually could happen. If you died at the same time when you reached a growth milestone, you'd actually be revived. Didn't change anything regarding this for now.